### DISCUSSION TITLE
Windows 11 Build Issues

### DISCUSSION THREAD

**Discussion by md-phantom** [AcademySoftwareFoundation/OpenRV#548] [Build Issues] [answered]
I am encountering a bunch of PySide + Python related compile errors, which are along the lines of the following:

C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\tuple(969,9): error G0546F524: static_assert failed "get<T>(const tuple<Types...>&) requires T to occur exactly
        once in Types. (N4971 [tuple.elem]/5)" [D:\dev\cmake-sources\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]
   169>C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\tuple(981,9): error G6D9AF971: static_assert failed "get<T>(tuple<Types...>&&) requires T to occur exactly once
        in Types. (N4971 [tuple.elem]/5)" [D:\dev\cmake-sources\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]
   169>C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\tuple(993,9): error G44F83FE2: static_assert failed "get<T>(const tuple<Types...>&&) requires T to occur exactl
       y once in Types. (N4971 [tuple.elem]/5)" [D:\dev\cmake-sources\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]

         (multimediawidgets) Clang: 4 diagnostic messages:
   169>C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\tuple(957,9): error G9DD9CECB: static_assert failed "get<T>(tuple<Types...>&) requires T to occur exactly once
       in Types. (N4971 [tuple.elem]/5)" [D:\dev\cmake-sources\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]
            C:/msys64/tmp/QtMultimediaWidgets_global_lKawYF.hpp:1:10: note: in file included from C:/msys64/tmp/QtMultimediaWidgets_global_lKawYF.hpp:1:
            D:/dev/cmake-sources/OpenRV/_build/_deps/rv_deps_pyside2-src/pyside3_build/py3.10-qt5.15.2-64bit-release/pyside2/PySide2/QtMultimediaWidgets_global.h:51:10: note: in file included from D:/dev/cmak
         e-sources/OpenRV/_build/_deps/rv_deps_pyside2-src/pyside3_build/py3.10-qt5.15.2-64bit-release/pyside2/PySide2/QtMultimediaWidgets_global.h:51:
            C:/Qt/5.15.2/msvc2019_64/include/QtCore/qnamespace.h:43:10: note: in file included from C:/Qt/5.15.2/msvc2019_64/include/QtCore/qnamespace.h:43:
            C:/Qt/5.15.2/msvc2019_64/include/QtCore/qglobal.h:1307:10: note: in file included from C:/Qt/5.15.2/msvc2019_64/include/QtCore/qglobal.h:1307:
            C:/Qt/5.15.2/msvc2019_64/include/QtCore/qglobalstatic.h:101:10: note: in file included from C:/Qt/5.15.2/msvc2019_64/include/QtCore/qglobalstatic.h:101:
            C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\mutex:18:10: note: in file included from C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\
         MSVC\14.41.34120\include\mutex:18:
            C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\thread:18:10: note: in file included from C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools
         \MSVC\14.41.34120\include\thread:18:

For the life of me, I really can't figure out what the issue is here. For reference, I am using VS 2022, with the Windows 11 SDK 10.0.26100.0 and MSVC v143 (v14.38-17.8).

This really looks like the most fiddly piece of software to build

**Comment by CraigMason** [comment]
This looks like the same issue as here: https://github.com/AcademySoftwareFoundation/OpenRV/issues/549

**Comment by cedrik-fuoco-adsk** [comment, marked as answer]
I'll copy my comments from the issue here:

For now, I would suggest to download to Microsoft Visual Studio 2022 version 17.10 if you need to compile OpenRV right away. You can find older versions of the installer here: https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-history#release-dates-and-build-numbers

![Screenshot from 2024-09-03 08-37-34](https://github.com/user-attachments/assets/449bf2af-c57d-4f07-a055-0e240a8811b3)
